src/hint_processor/math_utils.zig

```diff
@@ -0,0 +1,685 @@
+
+    // Main logic: ids.is_250 = 1 if ids.addr < 2**250 else 0
+    const is_250 = if (addr.numBits() <= 250) Felt252.one() else Felt252.zero();
+
+    try hint_utils.insertValueFromVarName(allocator, "is_250", MaybeRelocatable.fromFelt(is_250), vm, ids_data, ap_tracking);
+}
+
+// Implements hint:
+//   PRIME = 2**255 - 19
+//   II = pow(2, (PRIME - 1) // 4, PRIME)
+
+//   xx = ids.xx.low + (ids.xx.high<<128)
+//   x = pow(xx, (PRIME + 3) // 8, PRIME)
+//   if (x * x - xx) % PRIME != 0:
+//       x = (x * II) % PRIME
+//   if x % 2 != 0:
+//       x = PRIME - x
+//   ids.x.low = x & ((1<<128)-1)
+//   ids.x.high = x >> 128
+//   Note: doesnt belong to and is not variation of any hint from common/math
+pub fn splitXx(
+    allocator: std.mem.Allocator,
+    vm: *CairoVM,
+    ids_data: std.StringHashMap(HintReference),
+    ap_tracking: ApTracking,
+) !void {
+    const xx_u = try Uint256.fromVarName("xx", vm, ids_data, ap_tracking);
+    const x_addr = try hint_utils.getRelocatableFromVarName("x", vm, ids_data, ap_tracking);
+
+    const xx = xx_u.low.toInteger() + xx_u.high.mul(Felt252.pow2Const(128)).toInteger();
+
+    var x = field_helper.powModulus(
+        xx,
+        @divFloor(SPLIT_XX_PRIME + 3, 8),
+        SPLIT_XX_PRIME,
+    );
+
+    if (@mod(x * x - xx, SPLIT_XX_PRIME) != 0)
+        x = @mod(x * II, SPLIT_XX_PRIME);
+
+    if (@mod(x, 2) != 0)
+        x = SPLIT_XX_PRIME - x;
+
+    try vm.insertInMemory(
+        allocator,
+        x_addr,
+        MaybeRelocatable.fromFelt(Felt252.fromInt(u512, x & std.math.maxInt(u128))),
+    );
+    try vm.insertInMemory(allocator, try x_addr.addUint(1), MaybeRelocatable.fromFelt(Felt252.fromInt(u512, x >> 128)));
+}
+
+const SPLIT_XX_PRIME: u256 = 57896044618658097711785492504343953926634992332820282019728792003956564819949;
+const II: u256 = 19681161376707505956807079304988542015446066515923890162744021073123829784752;
+
+pub fn isQuadResidue(
+    allocator: std.mem.Allocator,
+    vm: *CairoVM,
+    ids_data: std.StringHashMap(HintReference),
+    ap_tracking: ApTracking,
+) !void {
+    const x = try hint_utils.getIntegerFromVarName("x", vm, ids_data, ap_tracking);
+
+    if (x.isZero() or x.equal(Felt252.one())) {
+        try hint_utils.insertValueFromVarName(allocator, "y", MaybeRelocatable.fromFelt(x), vm, ids_data, ap_tracking);
+        // } else if Pow::pow(felt_to_biguint(x), &(&*CAIRO_PRIME >> 1_u32)).is_one() {
+    } else if (x.pow((try Felt252.Max.divRem(Felt252.two())).q.toInteger()).equal(Felt252.one())) {
+        try hint_utils.insertValueFromVarName(allocator, "y", MaybeRelocatable.fromFelt(x.sqrt() orelse Felt252.zero()), vm, ids_data, ap_tracking);
+    } else {
+        try hint_utils.insertValueFromVarName(
+            allocator,
+            "y",
+            MaybeRelocatable.fromFelt((try x.div(Felt252.three()))
+                .sqrt() orelse Felt252.zero()),
+            vm,
+            ids_data,
+            ap_tracking,
+        );
+    }
```

Comment:
Here as the only thing that changes is the `MaybeRelocatable` argument, can we create a variable for that and just call the `insertValueFromVarName` method outside of the if statement?